modules/core/04-channel/types/packet.go

```diff
@@ -195,6 +196,9 @@ func (p Packet) ValidateBasic() error {
 	if len(p.Data) == 0 {
 		return errorsmod.Wrap(ErrInvalidPacket, "packet data bytes cannot be empty")
 	}
+	if p.AppVersion != "" && slices.Contains([]IBCVersion{IBC_VERSION_UNSPECIFIED, IBC_VERSION_1}, p.ProtocolVersion) {
+		return errorsmod.Wrap(ErrInvalidPacket, "app version cannot be specified when packet uses protocol version 1")
```

Comment:
Just to cover the case in the error message where protocol version is empty.

```suggestion
		return errorsmod.Wrapf(ErrInvalidPacket, "app version cannot be specified when packet does not use protocol %s", IBC_VERSION_2)
```